src/azure-cli/azure/cli/command_modules/acs/tests/latest/test_aks_commands.py

```diff
@@ -5640,3 +5640,29 @@ def test_aks_create_with_SP_then_update_to_user_assigned_identity(self, resource
         # delete
         self.cmd(
             'aks delete -g {resource_group} -n {name} --yes --no-wait', checks=[self.is_empty()])
+
+    @AllowLargeResponse()
+    @AKSCustomResourceGroupPreparer(random_name_length=17, name_prefix='clitest', location='westus2')
+    def test_aks_disable_local_accounts(self, resource_group, resource_group_location):
+        aks_name = self.create_random_name('cliakstest', 16)
+        self.kwargs.update({
+            'resource_group': resource_group,
+            'name': aks_name,
+            'ssh_key_value': self.generate_ssh_keys()
+        })
+
+        create_cmd = 'aks create --resource-group={resource_group} --name={name} --ssh-key-value={ssh_key_value} --enable-managed-identity --disable-local-accounts'
+        self.cmd(create_cmd, checks=[
+            self.check('provisioningState', 'Succeeded'),
+            self.check('disableLocalAccounts', True)
+        ])
+
+        # update to enable local accounts
+        self.cmd('aks update --resource-group={resource_group} --name={name} --enable-local-accounts', checks=[
```

Comment:
better to add some "real validation" to check the local accounts is indeed disabled?

Reply:
I think that should be server behavior's validation, should be covered by something like AKS E2E test, and is out of CLI test scope?

Reply:
When we create an cluster using option `--disable-local-accounts` by CLI, the CLI's responsibility is to create a cluster with property `disableLocalAccounts: true`, that is the CLI successfully accepts the option and send the request to server and server creates a cluster with desired property. That's all of our expectation for client. As for whether the server respects `disableLocalAccounts: true` when getting credential, it is not related to client. Say one day AKS server has a bug, the static credential can still be fetched even  `disableLocalAccounts: true`, that should not affect CLI's test. If we add server behavior check in CLI test, the server bug will block CLI and CLI has nothing to do with that.

Reply:
yeah, then is there any E2E to cover this in "server" side?

Reply:
synced offline. norshtein will file one work item to track this.